src/ragapp/backend/middlewares/rate_limit.py

```diff
@@ -35,7 +55,7 @@ def _get_time_frame():
     return time.strftime("%Y-%m-%d")
 
 
-def _extract_user_id_from_request(request: Request) -> str:
+def _extract_user_info_from_request(request: Request) -> UserInfo:
```

Comment:
once moved to UserInfo, how about splitting this into two functions, separating the concerns: retrieving JWT token from cookie and parsing UserInfo from JWT token

Reply:
also i just realized that we don't validate the JWT token - it's for a later PR, I added an issue: https://github.com/ragapp/ragapp/issues/145

Reply:
Got it! I focused on getting the rate limit set up first, and I’ll add JWT validation later.